src/util/FileSystem.ts

```diff
@@ -24,14 +24,45 @@ export class FileReadError extends Error {
 export default class FileSystem {
 
     /**
-     * The directory to which all files should be written to.
+     *
+     * Resolves with the full path to the file written, else, rejects with a FileWriteError
+     *
+     * @param fileName  of the TypeScript file to be written, does not need to end in .ts
+     * @param path      path to the directory where the file should be written
+     * @param content   as string
      */
-    private readonly rootDir: string;
-
-    constructor(rootDir: string) {
-        this.rootDir = rootDir;
+    public async generateFile(fileName: string, path: string, content: string): Promise<string> {
+        const fullPath: string = this.createFullPath(fileName, path);
+        try {
+            await fs.ensureDir(path);
+            await fs.writeFile(fullPath, content);
+            return fullPath;
```

Comment:
If you want to return a promise here, you probably want to do:
`return Promise.resolve(fullPath); `
since fullPath is a string

and similarly in the catch rather than throwing:
`return Promise.reject(new FileWriteError(msg))`

Reply:
`async/await` implicitly returns a `Promise`

Reply:
I see- thanks for the explanation!